src/components/Board.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect, useState } from 'react'
+import Square from './Square'
+import { checkWinner, checkStale } from '../utils'
+import Confetti from 'react-dom-confetti'
+import confettiConfig from './confetti-config'
+import Restart from './Restart'
+
+//component, destructured props of type BoardTypeProps
+const Board = ({
+  gameHistory,
+  addGameHistory,
+  currentTurn,
+}: BoardTypeProps) => {
+  const [gameState, setGameState] = useState<GameState>(gameHistory[0])
+  const [isWinner, setIsWinner] = useState<boolean>(false)
+  const [isStale, setIsStale] = useState<boolean>(false)
+  const [winningSquares, setWinningSquares] = useState<number[]>([])
+  const [sliderValue, setSliderValue] = useState<number>(0)
+
+  /**
+   *
+   * @param gameState
+   * @param squareIndex
+   * @param newValue
+   * @returns
+   */
+  const updateBoardState = (
```

Comment:
I would put this into it's own file since it doesn't depend on anything in the React lifecycle. Only define functions in the React lifecycle if absolutely necessary